libgnucash/engine/gnc-commodity.cpp

```diff
@@ -326,13 +326,11 @@ gnc_quote_source_init_tables (void)
 
     for (i = 0; i < num_single_quote_sources; i++)
     {
-        single_quote_sources[i].type = SOURCE_SINGLE;
         single_quote_sources[i].index = i;
```

Comment:
A further potential refactoring would be to get rid of this `index`. I went through the code and as far as I can tell it's not used in any meaningful way within gnc-commodity.cpp and only exposed publicly via `gnc_quote_source_get_index`.

That function is only consumed once in `gnc_ui_commodity_update_quote_info` in dialog-commodity.c.

The usage there is a poor man's way to keep track of which quote source to activate in a listbox. The more Gtk idiomatic way would be to keep a pointer to each source quote directly in the gtk_liststore and search for that one when you need to figure out which line to activate. In theory searching for a line in the list store each time would be slower than accessing it by index. I believe using an index to be a premature optimization: the list of quote sources is small and there is only a single potential search while loading the dialog.

Reply:
More design consideration: if refactoring dialog-commodity.c is outside of your current scope, you could at the interim also rewrite `gnc_quote_source_get_index`, to calculate the index on the fly. It would have a similar impact on performance, yet eliminate this init routine.

Note that there is a slight difference between these two approaches. The current implementation and the second proposed rewrite build on the assumption that order of quote sources in the internal tables in gnc-commodity.c[pp] and as presented in dialog-commodity.c is equal and can't change. The first proposed alternative doesn't have that assumption. It instead only assumes the quote sources loaded when creating the dialog continue to exist until the dialog is closed again. That would leave room to reorder the quote sources in the list if we would want to. That difference really only matters if the dialog would be refactored at some point in the future though.